### ISSUE TITLE
Question on Training FSA-S

### ISSUE THREAD

**Issue by ghost** [KAIST-VICLab/U-Know-DiffPAN#1]
Dear Sungpyo Kim,
First of all, I would like to express my appreciation for your excellent work. I have a small question regarding the training process. I re-trained PriorNet and FSA-T on the GF2 dataset, and used the command python main.py --stage FSA_T --mode save to store the intermediate results required for distillation. However, when distilling FSA-S, the model does not converge to a satisfactory result even after 400 epochs. I would be very grateful if you could kindly advise me on what might be causing this issue.

2025-09-08 17:04:04 - INFO - ---diffusion result---25 [00:23<00:00,  1.04it/s]
2025-09-08 17:04:04 - INFO - {'SAM': 0.9613745331764221, 'ERGAS': 0.8010575294494628, 'PSNR': 41.195677757263184, 'CC': 0.9916585981845856, 'SSIM': 0.9785198271274567}
2025-09-08 17:04:04 - INFO - save model
2025-09-08 17:04:04 - INFO - saved performances
Epoch 420: 100%|████████████████████████| 620/620 [05:25<00:00,  1.90 batch/s]
2025-09-08 17:07:17 - INFO - [iter 260400/300000: d_lr  0.000025] - prenetwork loss 0.000000 denoise loss 0.005140 
Epoch 421: 100%|████████████████████████| 620/620 [04:57<00:00,  2.09 batch/s]
2025-09-08 17:12:14 - INFO - [iter 261020/300000: d_lr  0.000025] - prenetwork loss 0.000000 denoise loss 0.005133 
Epoch 422: 100%|████████████████████████| 620/620 [05:00<00:00,  2.06 batch/s]
2025-09-08 17:17:15 - INFO - [iter 261640/300000: d_lr  0.000025] - prenetwork loss 0.000000 denoise loss 0.007240 
Epoch 423: 100%|████████████████████████| 620/620 [05:00<00:00,  2.06 batch/s]
2025-09-08 17:22:15 - INFO - [iter 262260/300000: d_lr  0.000025] - prenetwork loss 0.000000 denoise loss 0.006158 
ddim sampling loop time step: 100%|███████████| 25/25 [00:23<00:00,  1.07it/s]
2025-09-08 17:24:39 - INFO - ---diffusion result---25 [00:23<00:00,  1.03it/s]
2025-09-08 17:24:39 - INFO - {'SAM': 0.9617240250110626, 'ERGAS': 0.8009722411632538, 'PSNR': 41.21452121734619, 'CC': 0.9917471617460251, 'SSIM': 0.9785892397165299}
2025-09-08 17:24:39 - INFO - save model

**Comment by yangyingkun** [comment]
你好，我在训练Priornet时，损失值一直为负数，这大概是由什么原因导致的呢？
Epoch 189 Iter [57456/300000], Train Loss: -4.110886

**Comment by wshaow** [comment]
> 你好，我在训练Priornet时，损失值一直为负数，这大概是由什么原因导致的呢？ Epoch 189 Iter [57456/300000], Train Loss: -4.110886

你好，我也遇到了相同的问题。按照论文里的公式(4)我給损失函数加了绝对值。
def beta_nll_loss(mean, variance, target, beta = 0.0):
    loss = 0.5 * ((target - mean) ** 2 / variance + variance.log())

    if beta > 0:
        loss = loss * variance.detach() ** beta

    return torch.abs(loss).mean()
